spec/xml_security_spec.rb

```diff
@@ -3,62 +3,57 @@
+    let(:base64_cert) { OpenSSL::X509::Certificate.new(begin_end_cert(saml_settings.certificate)) }
 
     subject do
-      XMLSecurity::SignedDocument.new(xml_string)
-    end
-
-    let(:base64cert) do
-      subject.document.at_xpath('//ds:X509Certificate', ds_namespace).text
+      request.send(:document).signed_document
     end
 
     describe '#validate_doc' do
       describe 'when softly validating' do
+        before do
+          allow(subject).to receive(:digests_match?).and_return false
+        end
+
         it 'does not throw NS related exceptions' do
-          expect(subject.validate_doc(base64cert, true)).to be_falsey
+          expect(subject.validate_doc(base64_cert_text, true)).to be_falsey
         end
 
-        context 'multiple validations' do
+        context 'with multiple validations' do
           it 'does not raise an error' do
-            expect { 2.times { subject.validate_doc(base64cert, true) } }.not_to raise_error
+            expect { 2.times { subject.validate_doc(base64_cert_text, true) } }.not_to raise_error
           end
         end
       end
 
-      describe 'when validating not softly' do
-        it 'throws NS related exceptions' do
-          expect do
-            subject.validate_doc(base64cert,
-                                  false)
-          end.to raise_error(SamlIdp::XMLSecurity::SignedDocument::ValidationError)
-        end
-
-        it 'raises Fingerprint mismatch' do
-          expect { subject.validate('no:fi:ng:er:pr:in:t', false) }.to(
-            raise_error(SamlIdp::XMLSecurity::SignedDocument::ValidationError,
-                        'Fingerprint mismatch')
-          )
-        end
+      describe 'when throwing errors' do
+        context 'when when the certs do not match' do
+          let(:wrong_cert) { xml_cert_text(custom_idp_x509_cert) }
 
-        it 'raises Digest mismatch' do
-          expect { subject.validate_doc(base64cert, false) }.to(
-            raise_error(SamlIdp::XMLSecurity::SignedDocument::ValidationError, 'Digest mismatch')
-          )
+          it 'raises key validation error' do
+            expect { subject.validate_doc(wrong_cert, false) }.to(
+              raise_error(
+                SamlIdp::XMLSecurity::SignedDocument::ValidationError,
+                'Key validation error'
+              )
+            )
+          end
         end
 
-        context 'Key validation error' do
-          let(:xml_string) do
-            xml = fixture('valid_response_sha1.xml')
-            xml.sub!('<ds:DigestValue>pJQ7MS/ek4KRRWGmv/H43ReHYMs=</ds:DigestValue>',
-              '<ds:DigestValue>b9xsAXLsynugg3Wc1CI3kpWku+0=</ds:DigestValue>')
+        context 'when the digests do not match' do
+          before do
+            allow(subject).to receive(:digests_match?).and_return false
           end
 
-          it 'raises Key validation error' do
-            expect { subject.validate_doc(base64cert, false) }.to(
+          it 'raises digests error' do
```

Comment:
(suggestion, non-blocking): Instead of `raises digests error`, use `raises digest mismatch error`

Reply:
updated in https://github.com/18F/saml_idp/pull/124/commits/8658a88a0e129e2a06f451b4e63e65c2dffa09ce